h/models/group.py

```diff
@@ -150,6 +167,40 @@ def _write_principal(group):
     }.get(group.writeable_by)
 
 
+class _GroupMatcher(object):
+    """Abstract base class for group matcher classes."""
+
+    def __eq__(self, other):
+        """Return True if other has the same access flags as this matcher."""
+        attrs = ('joinable_by', 'readable_by', 'writeable_by')
+        for attr in attrs:
+            self_attr = getattr(self, attr)
+            other_attr = getattr(other, attr, None)
+            if self_attr != other_attr:
+                return False
+        return True
+
+    def __ne__(self, other):
+        """Return True if other has different access flags than this matcher."""
+        return not self.__eq__(other)
+
+
+class OpenGroupMatcher(_GroupMatcher):
```

Comment:
Although the implementation of the `type` property does use these as matcher classes, `create_open_group()` and `create_private_group()` use them as containers for the access flag values. It could be that `OpenGroupAccessFlags` or something else would be a better name.